### PR TITLE
 Keep the newest cloud record per VM per month 

### PR DESCRIPTION
**Pull request by gregcorbett**
This is a halfway house between the pre-LRVM record loading and the LRVM record loading, where we keep multiple records for a VM (one per month) but take the newest record we receive per month (as opposed to doing some `MeasurementTime` base check).

This makes it easier for sites to republish because they don't need to worry about the state of the repository.

I have created #193 about the problems creating a test case around this.

### CODE REVIEW

File: schemas/cloud.sql [hunk 1]
```diff
@@ -119,7 +119,7 @@ BEGIN
         END IF;
     END IF;
 
-    INSERT INTO CloudRecords(RecordCreateTime, VMUUID, SiteID, CloudComputeServiceID, MachineName,
+    REPLACE INTO CloudRecords(RecordCreateTime, VMUUID, SiteID, CloudComputeServiceID, MachineName,
```
Comment (tofu-rocketry) [inline schemas/cloud.sql]: What is the benefit of using REPLACE here?

File: schemas/cloud.sql [hunk 2]
```diff
@@ -132,53 +132,6 @@ BEGIN
         suspendDuration, wallDuration, cpuDuration, cpuCount,
         networkType, networkInbound, networkOutbound, publicIPCount, memory, disk,
         benchmarkType, benchmark, storageRecordId, imageId, cloudType, DNLookup(publisherDN))
-      ON DUPLICATE KEY UPDATE
```
Comment (tofu-rocketry) [inline schemas/cloud.sql]: Nice to get this simplified.

File: schemas/cloud.sql (hunk 1 above)
Comment (gregcorbett) [reply schemas/cloud.sql]: The benefit is we don't have to think about how the "upsert" is handled, as we currently do with the `INSERT... ON DUPLICATE KEY UPDATE`, `REPLACE` just keeps the last one.

We could achieve the same with `INSERT... ON DUPLICATE KEY UPDATE`, but it wouldn't reduce the complexity of the `ON DUPLICATE KEY UPDATE` very much.

Comment (tofu-rocketry) [reply schemas/cloud.sql]: Ah, so we were limited to using `INSERT... ON DUPLICATE KEY UPDATE` before as we needed to do fiddly things with MeasurementTime, right? But now we're not limited to that so `REPLACE` is the simpler option?

Comment (gregcorbett) [reply schemas/cloud.sql]: 👍

**Comment by tofu-rocketry** [comment]
I've checked that the update script seems to create the same procedure definition as the base cloud schema.

**Comment by tofu-rocketry** [comment]
I've renamed the update script as the numbers it refers to should bridge the whole gap from the last time the schema was changed. Could of other minor changes.

**Pull request merged**